### PR TITLE
FEAT: Filter saved queries and connections

### LINKED ISSUE #2033: FEAT: Ability to filter connections and saved queries
It's possible to filter tables and schemas by typing in the filter box. This is useful when there are a lot of tables. But what if a user has a lot of saved queries, or saved connections?

This should support:
- Saved connection filtering (just filter the items in the store, like we do tables)
- Saved query filtering (same)

### PR DESCRIPTION
**Pull request by not-night-but**
resolve #2033

This allows the user to filter saved connections and saved queries:

Peek 2024-04-25 11-12
Peek 2024-04-25 11-13

### CODE REVIEW

**Comment by rathboma** [comment]
Looks great! Will merge after the next release - can you make sure this works in ultimate with workspaces?

**Comment by not-night-but** [comment]
@rathboma will do!

File: apps/studio/src/store/modules/data/query/LocalQueryModule.ts [hunk 1]
```diff
@@ -9,9 +9,35 @@ export const LocalQueryModule: DataStore<FavoriteQuery, DataState<FavoriteQuery>
     loading: false,
     error: null,
     pollError: null,
+    filter: undefined // maybe this can be more advanced? date filter?
   },
   mutations: mutationsFor<FavoriteQuery>({
     // more mutations go here
```
Comment (rathboma) [inline apps/studio/src/store/modules/data/query/LocalQueryModule.ts:15]: So looks like this isn't implemented for queries and connections that come from the cloud?

**Review by rathboma — COMMENTED** [review]
Is this workspace enabled?

**Comment by rathboma** [comment]
@not-night-but - does this work in ultimate with workspaces?

If so, then merge after fixing conflicts.

**Review by rathboma — APPROVED** [review]

**Pull request merged**